agent/config/types.go

```diff
@@ -210,3 +219,42 @@ func (data *SensitiveRawMessage) UnmarshalJSON(jsonData []byte) error {
 	data.contents = json.RawMessage(jsonData)
 	return nil
 }
+
+// DefaultBool makes it possible to understand if a variable was set explicitly or relies on a default setting
+type DefaultBool int
```

Comment:
Could this name something along the lines of `TriBool` ?
Could you move this above the `iota` ?

Reply:
How about moving the type and iota into their own file?

Reply:
I like the idea of making this more widely available